model/ActivityMonitoringService.php

```diff
@@ -237,31 +234,75 @@ protected function getNumberOfActiveUsers($role = null)
+        $deliveryMonitoringService = $this->getServiceManager()->get(DeliveryMonitoringService::SERVICE_ID);
+        $deliveryService = DeliveryAssemblyService::singleton();
 
-        $deliveries = DeliveryAssemblyService::singleton()->getAllAssemblies();
+        $limit = isset($params['rows']) ? $params['rows'] : 10;
+        $offset = isset($params['page']) ? ($params['page']-1) * $limit : 0;
+        $deliveries = $deliveryService->getRootClass()->getInstances(true, [
+            'order' => RDFS_LABEL,
+            'offset' => $offset,
+            'limit' => $limit
+        ]);
+
+        // deliveries count + retired deliveries row
+        $total = $deliveryService->getRootClass()->countInstances([], ['recursive' => true]) + 1;
+
+        $newResult = [];
         foreach ($deliveries as $delivery) {
             $newResult[$delivery->getUri()] = $statusesArray;
             $newResult[$delivery->getUri()]['label'] = $delivery->getLabel();
         }
 
-        foreach ($deliveryMonitoringService->find([], ['asArray'=>true], true) as $sessionData) {
-            $deliveryId = isset($newResult[$sessionData['delivery_id']]) ? $sessionData['delivery_id'] : self::FIELD_RETIRED_DELIVERIES;
-            $newResult[$deliveryId][$sessionData['status']]++;
+        $data = $deliveryMonitoringService->getDeliveriesCountedStatuses(array_keys($deliveries));
+
+        if (count($deliveries) < $limit) {
+            $data = array_merge($data, $deliveryMonitoringService->getRetiredDeliveriesCountedStatuses());
         }
-        return $newResult;
+
+        $retiredTitle = __('Retired Deliveries');
+        foreach ($data as $row) {
+            $delivery_id = $row['delivery_id'] ?: self::FIELD_RETIRED_DELIVERIES;
+            if (!isset($newResult[$delivery_id])) {
+                $newResult[$delivery_id] = $statusesArray;
+                $delivery = $this->getResource($delivery_id);
+                if (!$delivery->exists()) {
+                    $newResult[$delivery_id]['label'] = $retiredTitle;
+                }
+            }
+
+            $newResult[$delivery_id][$row['status']] += $row['cnt'];
+        }
+
+        $list = [];
+        foreach ($newResult as $key => $row) {
+            if ($row['label'] != $retiredTitle) {
+                $list[$key] = $row;
+            } else {
+                if (!isset($list[self::FIELD_RETIRED_DELIVERIES])) {
+                    $list[self::FIELD_RETIRED_DELIVERIES] = $row;
+                } else {
+                    foreach ($row as $status => $val) {
+                        if ($status == 'label') {
+                            continue;
+                        }
+                        $list[self::FIELD_RETIRED_DELIVERIES][$status] = (int)$list[self::FIELD_RETIRED_DELIVERIES][$status] + (int)$val;
+                    }
+                }
+            }
+        }
+
+        return ['data' => $list, 'total' => $total];
```

Comment:
Currently this function is supposed to be used with datatable. You have to return only `$list` variable and wrap it into `['data' => ..., $total => ...` array inside `DeliveriesActivityDatatable`